lib/backends/sdc/containers.js

```diff
@@ -681,43 +667,96 @@ function defaultFabricNetwork(opts, container, payload, callback) {
  * we will find and use the specified network.
  */
 function namedNetwork(opts, container, payload, callback) {
+    assert.object(opts, 'opts');
+    assert.object(opts.account, 'opts.account');
+    assert.string(opts.account.uuid, 'opts.account.uuid');
+    assert.object(opts.config, 'opts.config');
+    assert.object(opts.config.napi, 'opts.config.napi');
+    assert.object(opts.log, opts.log);
+    assert.object(container, 'container');
+    assert.object(container.HostConfig, 'container.HostConfig');
+    assert.string(container.HostConfig.NetworkMode,
+        'container.HostConfig.NetworkMode');
 
     // need to search on networks by: name, fabric-true, owner_uuid
     var log = opts.log;
     var query = container.HostConfig.NetworkMode;
     var napi = getNapiClient(opts.config.napi);
 
+    function logListResults(err, networks) {
+        log.debug({ results: networks, user: opts.account.uuid },
+            'Networks: Found %s results for query %s',
+            networks ? networks.length : 0, query);
+        callback(err, networks);
+        return;
+    }
+
+    // NOTE: the order of the functions in this parallel() call is significant;
+    // they are ordered by how docker prefers to resolve IDs:
+    // 1. exact id match
+    // 2. exact name match
+    // 3. partial id match
     vasync.parallel({
         funcs: [
+            function byExactId(cb) {
+                // length === 64, and 0..31 === 32..63, or it's an 'impossible'
+                // id under our double-uuid convention, and we can skip it.
+                if (query.substr(0, 32) !== query.substr(32)) {
+                    log.debug({ query: query },
+                        'Networks: impossible exactId: %s, skipping', query);
+                    setImmediate(cb, null, []);
```

Comment:
Ah OK, I thought this condition being true when that query is an invalid input, as in it's an error and we don't know what to do with it.

So is the condition:

```
(query.substr(0, 32) !== query.substr(32))
```

meant to filter out anything that is not a 64 characters Docker ID? If so, it seems it would have a few false negative, like if `query` was the empty string.